### PR TITLE
feat(email): email address helper class for formatting

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: docs/guides/helpers.rst [hunk 1]
```diff
@@ -48,4 +52,31 @@ Interface and annotations
 
 .. code:: php
 
-   elgg_view_entity_icon($user, 'small', array('use_hover' => false));
\ No newline at end of file
+   elgg_view_entity_icon($user, 'small', array('use_hover' => false));
+
+E-mail address formating
```
Comment (jdalsem) [inline docs/guides/helpers.rst]: formatting

File: docs/guides/upgrading.rst [hunk 2]
```diff
@@ -204,7 +204,7 @@ Removed classes/interfaces
  * ``ODD`` and all classes beginning with ``ODD*``.
  * ``XmlElement``
  * ``Elgg_Notifications_Event``: Use ``\Elgg\Notifications\Event``
- * ``Elgg\Mail\Address``: use ``Elgg\Email``
+ * ``Elgg\Mail\Address``: use ``Elgg\Email\Address``
```
Comment (jdalsem) [inline docs/guides/upgrading.rst]: huh... this is the same...

File: engine/classes/Elgg/Email.php [hunk 3]
```diff
@@ -5,20 +5,20 @@
 use ElggEntity;
 use ElggUser;
 use InvalidParameterException;
-use Zend\Mail\Address;
+use Elgg\Email\Address;
 
 /**
  * Email message
  */
 final class Email {
```
Comment (jdalsem) [inline engine/classes/Elgg/Email.php]: why final?

File: engine/classes/Elgg/Email.php [hunk 4]
```diff
@@ -247,38 +247,4 @@ protected static function prepareTo($to) {
...
-	/**
-	 * Parses strings like "Evan <evan@elgg.org>" into name/email objects.
-	 *
-	 * This is not very sophisticated and only used to provide a light BC effort.
-	 *
-	 * @param string $contact e.g. "Evan <evan@elgg.org>"
-	 *
-	 * @return Address
-	 */
-	public static function fromString($contact) {
```
Comment (jdalsem) [inline engine/classes/Elgg/Email.php]: do we need to document these functions have been moved? They were public right?

File: engine/classes/Elgg/Email.php [hunk 5]
```diff
@@ -247,38 +247,4 @@ protected static function prepareTo($to) {
...
-	 * Format an email address and name into a formatted email address
-	 *
-	 * eg "Some name <someone@example.com>"
-	 *
-	 * @param string $email the email address
-	 * @param string $name  the name
-	 *
-	 * @return string
-	 */
-	public static function getFormattedEmailAddress($email, $name = null) {
```
Comment (jdalsem) [inline engine/classes/Elgg/Email.php]: same

**Review by jdalsem — DISMISSED** [review]

File: engine/classes/Elgg/Email.php (hunk 3 above)
Comment (jeabakker) [reply engine/classes/Elgg/Email.php]: i don't know

File: engine/classes/Elgg/Email.php (hunk 5 above)
Comment (jeabakker) [reply engine/classes/Elgg/Email.php]: is documented as it came from a different class, and that change is listed

File: docs/guides/upgrading.rst (hunk 2 above)
Comment (jeabakker) [reply docs/guides/upgrading.rst]: no it's not ;)

**Pull request merged**